mods_cocina_mappings/mods_to_cocina_titleInfo.txt

```diff
@@ -0,0 +1,386 @@
+Titles
+
+1. Basic title
+How to ID: only subelement of titleInfo is title and no titleInfo type attribute
+<titleInfo>
+  <title>Gaudy night</title>
+</titleInfo>
+{
+  "title": [
+    {
+      "value": "Gaudy night"
+    }
+  ]
+}
+
+2. Title with parts
+How to ID: multiple subelements in titleInfo
+<titleInfo>
+  <nonSort>The</nonSort>
+  <title>journal of stuff</title>
+  <partNumber>volume 5</partNumber>
+  <partName>special issue</partName>
+</titleInfo>
+{
+"title": [
+    {
+      "structuredValue": [
+        {
+          "value": "The",
+          "type": "nonsorting characters"
+        },
+        {
+          "value": "journal of stuff",
+          "type": "main title"
+        },
+        {
+          "value": "volume 5",
+          "type": "part"
+        },
+        {
+          "value": "special issue",
+          "type": "part"
+        },
+        {
+          "note": [
+            {
+              "value": 4,

```

Comment:
This isn't in the original metadata, so why add it? Why put this in metadata and not just determine when needed?

Reply:
This was born out of a specific frustration with how mods_display handles the MODS nonSort element. It's hard-coded to add a space when concatenating nonSort with title, even though there should be exceptions to this rule (after L' or al-, for example). In current practice there is no way to indicate these exceptions in the metadata, so correct display is reliant on having the logic for the exceptions hard-coded into the display rules. Explicitly stating this value means that the code only has to compare it with the number of characters in the nonSort string to determine whether a space is required, rather than needing to know what the exceptions are. Same for determining the 245 second indicator when converting to MARC -- the mapping wouldn't need to specify the exceptions to calculate that value, because it is stated in the metadata.